### PR TITLE
Add spec for ComplexOp

### LINKED ISSUE #468: Add spec for ComplexOp

### PR DESCRIPTION
**Pull request by ghpvnist**
closes #468

### CODE REVIEW

**Comment by ghpvnist** [comment]
Verification is `revisit` since the verifier currently allows `lhs` and `rhs` to have different floating-point element type. Issue raised in #492

File: docs/spec_draft.md [hunk 1]
```diff
@@ -609,6 +610,44 @@ matrix, then the behavior is undefined.
...
+
+### Outputs
+
+| Name     | Type                   |
+|----------|------------------------|
+| `result` | tensor of complex type |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same floating-point element type.
```
Comment (sdasgup3) [inline docs/spec_draft.md:635]: How about element_type(`lhs`) $=$ element_type(`rhs`). As we already mentioned that the element type is floating point in the table.

**Review by sdasgup3 — APPROVED** [review]

File: docs/status.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ one of the following tracking labels.
 | clamp                    | no            | yes*         | yes*           | yes             | no          |
 | collective_permute       | no            | revisit      | revisit        | no              | no          |
 | compare                  | no            | yes*         | yes*           | yes             | no          |
-| complex                  | no            | yes*         | yes*           | yes             | no          |
+| complex                  | yes           | revisit      | yes            | yes             | no          |
```
Comment (burmako) [inline docs/status.md:66]: Given the discussion in #492, I think that Verification should be "yes".

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec_draft.md [hunk 3]
```diff
@@ -609,6 +610,44 @@ matrix, then the behavior is undefined.
...
+  * (C2) shape(`lhs`) $=$ shape(`rhs`) $=$ shape(`result`).
+  * (C3) element_type(`lhs`) $=$ underlying floating-point element type in
+    element_type(`result`).
```
Comment (burmako) [inline docs/spec_draft.md:638]: Also, not sure if it's just me, but it feels more natural to express C2 and C3 `with element_type(result)` on the left-hand side of the equation, i.e.:

```
  * (C2) `shape(result) = shape(lhs)`.
  * (C3) `element_type(result) = complex_type(element_type(lhs))`.
```

This way, it's also easier to implement type inference for this op based on the Constraints section. It's not hard to invert `element_type(lhs) = real_type(element_type(result))` - but it feels awkward. Given how many times I wrote "feels", this is not a blocking comment, and we can leave figuring this out to #289.

**Review by burmako — CHANGES_REQUESTED** [review]

**Review by burmako — APPROVED** [review]

**Pull request merged**